src/job.rs

```diff
@@ -736,26 +739,26 @@ impl Job {
             return Status::RetryLater;
         }
 
-        if let Some(sentbox_folder) = &context.get_config(Config::ConfiguredSentboxFolder).await {
-            job_try!(
-                imap.resync_folder_uids(context, sentbox_folder.to_string())
-                    .await
-            );
-        }
+        let configured_folders: Vec<_> = async_std::stream::from_iter(&[
+            Config::ConfiguredSentboxFolder,
+            Config::ConfiguredInboxFolder,
+            Config::ConfiguredMvboxFolder,
+        ])
+        .filter_map(|c| context.get_config(*c))
+        .collect()
+        .await;
 
-        if let Some(inbox_folder) = &context.get_config(Config::ConfiguredInboxFolder).await {
-            job_try!(
-                imap.resync_folder_uids(context, inbox_folder.to_string())
-                    .await
-            );
+        let all_except_configured =
+            job_try!(imap.list_folders_except(context, &configured_folders).await);
```

Comment:
It almost seems a little risky to me to use `job_try()` here because it will just fail the job instead of retrying it (return Finished(err) instead of RetryNow). And then the UIDs are not resynced, which sounds bad.

Did I oversee sth. in this chain of thought? If not, I'll change this

Reply:
Right, this should return `RetryLater` indeed. Maybe even keep resyncing other folders and only return `RetryLater` at the end of the job, so we accidentally rewrite UID with an UID from some unconfigured folder because another unconfigured folder failed and we didn't get to configured folders.